lib/awful/keygrabber.lua

```diff
@@ -27,11 +101,99 @@ local function grabber(mod, key, event)
     end
 end
 
+local function runner(self, modifiers, key, event)
+    -- Stop the keygrabber with the `release_key`
+    if (key == self.release_key or conversion[key] == self.release_key)
+        and event == self.release_event and self.release_key then
+        self:stop(key, modifiers)
+        return false
+    end
+
+    -- Stop when only a subset of keys are allowed and it isn't one of them.
+    if self._private.allowed_keys and not self._private.allowed_keys[key] then
+        self:stop(key, modifiers)
+        return false
+    end
+
+    -- Support multiple release keys
+    if type(self.release_key) == "table" then
+        for _, k in ipairs(self.release_key) do
+            if (k == key or conversion[key] == k) and event == self.release_event then
```

Comment:
I'd move the check for `release_event` out of the `for` loop up the `type(self.release_key) == "table"` since it's loop-invariant.